### PR TITLE
Add functionality for player to spawn balls, as well as improve smacking

### PR DESCRIPTION
**Pull request by khooroko**
**Description**
This PR allows the player to spawn the balls instead of having it spawn at random locations. At the moment, the player will be given 50 balls at the start to throw. Press the trigger to release the ball. For now, I use a black capsule in place of a hand. 
ezgif com-crop

Was thinking that a hand may not be the best since we'll need to eventually accommodate balls of different sizes. Perhaps we can use particle effects or transparency or some emission to indicate that the ball is pending spawning?

@lyhvictoria

**Impact**
If impact is minor, no need for the reviewer to check out the branch to test locally before approval
- [x] Major

**Risks**
The implementation here seems to make a lot of things have references on one another (hand to spawner and player) but should not be too big of a problem.

**Test Plan**
Spawn with one hand and smack with the other. You can toggle spawning hand (left or right) in the inspector for now, switching will be implemented another day

### CODE REVIEW

**Comment by lyhvictoria** [comment]
I think you should rename this PR, cos player spawning could mean spawning the player not the player spawning the balls

File: Assets/Scripts/SpawnerHand.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    // Takes the next ball out of the queue and into the hand
+    void DequeueNextBall() {
+        currentBall = ballsToThrow.Dequeue();
+        currentBall.GetComponent<Collider>().enabled = false;
+        currentBall.transform.parent = this.transform;
+        currentBall.transform.position = this.transform.position;
+        currentBall.GetComponent<Rigidbody>().isKinematic = true;
+        currentBall.SetActive(true);
+    }
+
```
Comment (lyhvictoria) [inline Assets/Scripts/SpawnerHand.cs:49]: You can also add an Enqueue function to add balls back into the pool, for when balls get into the goals and are destroyed, or when the balls maybe time out for being stationary for too long.

File: Assets/Scripts/Tool.cs [hunk 2]
```diff
@@ -1,15 +1,27 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using Valve.VR;
 
 public class Tool : MonoBehaviour {
     HapticFeedback hapticFeedback;
+    SteamVR_Behaviour_Pose hand;
+    public float forceMultiplier = 5;
```
Comment (lyhvictoria) [inline Assets/Scripts/Tool.cs:9]: Rmb to set this to private final in future after fine tuning.

File: Assets/Scripts/SpawnerHand.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+using Valve.VR;
+
+// Hand for spawning balls. Set to active only when there is a ball in the queue for this player. Otherwise, a Tool should
+// be active on the player's hand instead.
+public class SpawnerHand : MonoBehaviour {
+    // A queue of balls that this player should throw. The balls are to be added as children to this GameObject but inactive.
+    public Queue<GameObject> ballsToThrow;
+    public SteamVR_Action_Boolean click = SteamVR_Input.GetAction<SteamVR_Action_Boolean>("InteractUI");
+    public float throwingForce = 100;
+    public float spawnDelay = 0.25f;
```
Comment (lyhvictoria) [inline Assets/Scripts/SpawnerHand.cs:13]: Rmb to set this to private final in future after fine tuning.

File: Assets/Scripts/SpawnerHand.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+using UnityEngine;
+using Valve.VR;
+
+// Hand for spawning balls. Set to active only when there is a ball in the queue for this player. Otherwise, a Tool should
+// be active on the player's hand instead.
+public class SpawnerHand : MonoBehaviour {
+    // A queue of balls that this player should throw. The balls are to be added as children to this GameObject but inactive.
+    public Queue<GameObject> ballsToThrow;
+    public SteamVR_Action_Boolean click = SteamVR_Input.GetAction<SteamVR_Action_Boolean>("InteractUI");
+    public float throwingForce = 100;
```
Comment (lyhvictoria) [inline Assets/Scripts/SpawnerHand.cs:12]: Rmb to set this to private final in future after fine tuning.

**Review by lyhvictoria — APPROVED** [review]
Some comments, but they can be addressed in future PRs.

File: Assets/Scripts/Tool.cs (hunk 2 above)
Comment (khooroko) [reply Assets/Scripts/Tool.cs:9]: it's public so that it can be adjusted within the inspector

File: Assets/Scripts/SpawnerHand.cs (hunk 1 above)
Comment (khooroko) [reply Assets/Scripts/SpawnerHand.cs:49]: yup, will be in the Ball script eventually

File: Assets/Scripts/Tool.cs (hunk 2 above)
Comment (lyhvictoria) [reply Assets/Scripts/Tool.cs:9]: yes, i know but once its been fine tuned, we don't want to accidentally change it. It works for now.

**Pull request merged**